templates/admin/ppom-fields.php

```diff
@@ -55,60 +55,395 @@
 
 $product_id = isset( $_GET['product_id'] ) ? intval( $_GET['product_id'] ) : '';
 
+$license_data   = get_option( 'ppom_pro_license_data', array() );
+$plan_category  = NM_PersonalizedProduct_Admin::LICENSE_PLAN_FREE;
+$license_status = ! empty( $license_data->license ) ? $license_data->license : 'invalid';
+
+if ( isset( $license_data->plan ) && is_numeric( $license_data->plan ) ) {
+	$plan_category = NM_PersonalizedProduct_Admin::get_license_category( intval( $license_data->plan ) );
+}
+
+$fields_groups = [
+	'text-group' => [
+		'label' => __( 'Text Inputs', 'woocommerce-product-addon' ),
+		'fields' => [
+			[
+				'slug'        => 'text',
+				'title'       => __( 'Text Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Simple text field', 'woocommerce-product-addon' ),
+				'icon'        => '<i class="fa fa-pencil-square-o" aria-hidden="true"></i>'
+			],
+			[
+				'slug'        => 'textarea',
+				'title'       => __( 'Textarea Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Simple area field', 'woocommerce-product-addon' ),
+				'icon'        => '<i class="fa fa-file-text-o" aria-hidden="true"></i>'
+			],
+			[
+				'slug'        => 'email',
+				'title'       => __( 'Email Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Simple email field', 'woocommerce-product-addon' ),
+				'icon'        => '<i class="fa fa-user-plus" aria-hidden="true"></i>'
+			],
+			[
+				'slug'        => 'hidden',
+				'title'       => __( 'Hidden Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Simple hidden field', 'woocommerce-product-addon' ),
+				'icon'        => '<i class="fa fa-hashtag" aria-hidden="true"></i>'
+			],
+			[
+				'slug'        => 'phone',
+				'title'       => __( 'Phone Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Simple Phone field', 'woocommerce-product-addon' ),
+				'plan'        => NM_PersonalizedProduct_Admin::LICENSE_PLAN_1,
+				'icon'        => '<i class="fa fa-check" aria-hidden="true"></i>',
+			],
+			[
+				'slug'        => 'chained',
+				'title'       => __( 'Chained Input', 'woocommerce-product-addon' ),
+				'description' => __( 'Chained input field', 'woocommerce-product-addon' ),
+				'plan'        => NM_PersonalizedProduct_Admin::LICENSE_PLAN_1,
+				'icon'        => '<i class="fa fa-check" aria-hidden="true"></i>',
+			],
+			[
+				'slug'        => 'textcounter',
+				'title'       => __( 'Text Counter', 'woocommerce-product-addon' ),
+				'description' => __( 'Character count for input fields', 'woocommerce-product-addon' ),
+				'plan'        => NM_PersonalizedProduct_Admin::LICENSE_PLAN_1,
+				'icon'        => '<i class="fa fa-comments-o" aria-hidden="true"></i>',
```

Comment:
To avoid the repetitive nature of this element, you can use it for just the icon class, and have the syntax in the content, given that we only use FontAwesome and no other icons.